disnake/shard.py

```diff
@@ -330,12 +333,41 @@ class AutoShardedClient(Client):
     if TYPE_CHECKING:
         _connection: AutoShardedConnectionState
 
+    @overload
     def __init__(
-        self, *args: Any, loop: Optional[asyncio.AbstractEventLoop] = None, **kwargs: Any
-    ) -> None:
-        kwargs.pop("shard_id", None)
-        self.shard_ids: Optional[List[int]] = kwargs.pop("shard_ids", None)
-        super().__init__(*args, loop=loop, **kwargs)
+        self,
+        *,
+        asyncio_debug: bool = False,
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        shard_ids: List[int] = None,  # instead of Client's shard_id: Optional[int]
+        shard_count: Optional[int] = None,
+        enable_debug_events: bool = False,
+        connector: Optional[aiohttp.BaseConnector] = None,
+        proxy: Optional[str] = None,
+        proxy_auth: Optional[aiohttp.BasicAuth] = None,
+        assume_unsync_clock: bool = True,
+        max_messages: Optional[int] = 1000,
+        application_id: Optional[int] = None,
+        heartbeat_timeout: float = 60.0,
+        guild_ready_timeout: float = 2.0,
+        allowed_mentions: Optional[AllowedMentions] = None,
+        activity: Optional[BaseActivity] = None,
+        status: Optional[Union[Status, str]] = None,
+        intents: Intents = None,
+        chunk_guilds_at_startup: Optional[bool] = None,
+        member_cache_flags: MemberCacheFlags = None,
```

Comment:
```suggestion
        member_cache_flags: Optional[MemberCacheFlags] = None,
```

Reply:
Fixed in 83ddb4a3785dc78c0d97c8c33f77c5c6dbcff996.